### PR TITLE
Publish zeroconf service

### PR DESCRIPTION
**Pull request by raman325**
The server will now advertise itself via DNS-SD. This can be used downstream to support discovery in non-supervised installations of HA

### CODE REVIEW

File: src/lib/server.ts [hunk 1]
```diff
@@ -419,6 +433,19 @@ export class ZwavejsServer extends EventEmitter {
     await once(this.server, "listening");
     this.emit("listening");
     this.logger.info(`ZwaveJS server listening on ${localEndpointString}`);
+    this.responder = getResponder();
+    this.service = this.responder.createService({
+      name: this.driver.controller.homeId!.toString(),
```
Comment (balloob) [inline src/lib/server.ts:438]: Can we make this opt-in/opt-out ?

**Review by balloob — APPROVED** [review]

File: src/lib/server.ts (hunk 1 above)
Comment (raman325) [reply src/lib/server.ts:438]: yep, which do you prefer? opt in? we can add a new flag for it

Comment (raman325) [reply src/lib/server.ts:438]: what's your thought behind that btw? given we are hard open to the network, I can't imagine it's a security concern

Comment (balloob) [reply src/lib/server.ts:438]: It's causing extra traffic that is not needed in certain circumstances (ie the HA add-on).

We can make it opt-out.

**Pull request merged**